### DISCUSSION TITLE
[Feature Request] Linux Version

### DISCUSSION THREAD

**Discussion by ghost** [baffler/Transparent-Twitch-Chat-Overlay#52] [Ideas]
I am a Linux gamer and I don't know any other program.

**Comment by darkralts** [comment]
same, this program looks so nice, i would love this on linux as well

i dont think it would be too hard to make dotnet work on linux with mono, but im not a coder so i have no idea

**Comment by darkralts** [comment]
https://github.com/danieloneill/TwitchOverlay
this is the closest program i could find that has a linux version, im gonna try it out and see if it works for me, but i dont really like the look of it, and this program has exactly what im looking for :D looks like the OG twitch chat 
edit: also that other program i linked hasnt had an update since 2018 so im not sure if itl even work lole

**Comment by ghost** [comment]
If it works can you tell me how you installed it?
Cause I never install programs that are not available in my package manager,so I don't know how.
And is it actually better than using an transparent window of the twitch chat in the web?
Cause if not I will just use the web version.
But even if it's good,without updates or forks it isn't possible to add cool features.

**Comment by darkralts** [comment]
yeah, i never managed to get this one to work either unfortunately, it might work for you but basically what i did was download and extract it, and i ran the executable through the terminal, which you can do by ./path-to-file

**Comment by darkralts** [comment]
although, if youre still looking for a solution, i found something called restream, while i dont really like it it seems like the only option i have for right now sadly, heres a link
https://restream.io/

if mr owner man is reading this, i would greatly appreciate this port as im sure many other linux twitch streamers would aswell :D

**Comment by ghost** [comment]
For Restream,is it open source?
Cause if not,I won't use it.

**Comment by pjamroziak** [comment]
Hi @Toadfield, @darkralts ,
Restream.io isn't an open-source project.

TTCO (short repo's name) uses a lot of Windows libraries, unfortunately, create port isn't available by change a few lines of codes.

**Comment by darkralts** [comment]
Aw that's unfortunate to hear, thanks for the reply though :D and yeah,
unfortunately restream is proprietary, it's the only workable solution I
found for this sort of thing, maybe I'll take the plunge and learn coding
and make my own, I highly doubt I could even get to that level, but I guess
I could at least try :D

**Comment by Konpaku-LVX** [comment]
https://github.com/dongy7/twitch-chat-cli

**Comment by ghost** [comment]
@mrpajax Thanks for your effort and answering.

**Comment by ghost** [comment]
@autoteelar What is with that?
That just looks like the normal twitch web chat it cli.
Can you do cool things like viewing this over fullscreen applications and change the size of text etc. fastly or is it just a generic cli  twitch chat viewer?
Even if it has all the cool features,cli based twitch chat isn't as cool as displaying the normal witch chat,cause (correct me if I'm wrong) emotes won't looks as good as on normal twitch chat and you can't see raids and the spinning bit animation from donations,or can you?
Ans also can you ban/timeout people easily with that chat (I mean when doing that,then just switching to a virtual desktop with the normal browser twitch chat would make much more sense,but it would still be cool to do that)?

**Reply by MarceloRTonon** [reply]
well, you can make the terminal less opaque and it is our best shot right now. It is not perfect. I don't know... I would love to find something functional for this...

**Comment by darkralts** [comment]
You can do all that through i3 and terminal configs, I'll try this program
later as I'm pretty interested

**Comment by darkralts** [comment]
Well idk about banning and all that but I'll try it later and see what it
can do

**Comment by ghost** [comment]
ok,good luck

**Comment by Greenteally** [comment]
Still no support for linux?
